### PR TITLE
Initial FreeBSD support

### LINKED ISSUE #29: Add support for Freebsd

### PR DESCRIPTION
**Pull request by vkrasnov**
Implement initial FreeBSD support.

Fixes #29

### CODE REVIEW

File: src/device/mod.rs [hunk 1]
```diff
@@ -8,16 +8,16 @@ pub mod drop_privileges;
 mod integration_tests;
 pub mod peer;
 
-#[cfg(any(target_os = "macos", target_os = "ios"))]
+#[cfg(any(target_os = "macos", target_os = "freebsd"))]
```
Comment (ashleygwilliams) [inline src/device/mod.rs]: was that `target_os = "ios"` not load bearing?

File: src/device/mod.rs [hunk 2]
```diff
@@ -8,16 +8,16 @@ pub mod drop_privileges;
...
 pub mod poll;
 
 #[cfg(target_os = "linux")]
 #[path = "epoll.rs"]
 pub mod poll;
 
-#[cfg(any(target_os = "macos", target_os = "ios"))]
-#[path = "tun_darwin.rs"]
+#[cfg(any(target_os = "macos", target_os = "freebsd"))]
+#[path = "tun_bsd.rs"]
```
Comment (ashleygwilliams) [inline src/device/mod.rs]: we may want this file name to indicate that it's both darwin and bsd

Comment (vkrasnov) [reply src/device/mod.rs]: darwin is considered bsd

File: src/device/mod.rs (hunk 1 above)
Comment (vkrasnov) [reply src/device/mod.rs]: No, I'm afraid ios is never gonna support `utun` :(

**Comment by myfreeweb** [comment]
Hi, any updates on this? Looks like there's merge conflicts now..

**Comment by vkrasnov** [comment]
> Hi, any updates on this? Looks like there's merge conflicts now..

I plan to have official FreeBSD support by the end of this year

**Pull request closed without merging**